_episodes/02-shell_to_cwl.md

```diff
@@ -27,792 +27,215 @@ keypoints:
-~~~
-cwlVersion: v1.2
-class:
-~~~
-{: .language-yaml}
-The `cwlVersion` string defines which standard of the language is required for the tool or workflow. The most recent version is v1.2, and defaulting to this will enable your scripts to use all versions of the language, though some workflow engines which are not up-to-date may not run the script. This is, however, a hurdle to be tackled when you reach it.
-
-The `class` field defines what this particular script is. The majority of CWL scripts will fall into one of two classes: `CommandLineTool`, or `Workflow`. The former class is used for describing the interface for a command-line tool, while the latter class is used for collecting those tool descriptions into a workflow. In this lesson we will learn the differences between these two classes, how to pass data to and from command-line tools and specify working environments for these, and finally how to use a tool descriptor within a workflow.
-
-
-
-
-## Tool Descriptors
+CWL workflows are defined in a YAML script. This [quick tutorial](https://www.commonwl.org/user_guide/yaml/) explains the YAML script used in CWL. A CWL script contains the workflow and the requirements for running that workflow. All CWL scripts should start with two lines of code:
 
-
-To demonstrate the basic requirements for a tool descriptor we will recreate the standard hello world example. __Note: replace this with a domain specific (but similar complexity) example!__ This is the shell `echo` command that we will use:
-~~~
-$ echo 'hello World!'
-~~~
-{: .language-bash}
-~~~
-hello World!
-~~~
-{: .output}
-
-Create a file, `echo.cwl`, to contain your CWL example for this:
 ~~~
 cwlVersion: v1.2
-class: CommandLineTool
-
-baseCommand: [echo, 'hello World!']
+class: 
 ~~~
 {: .language-yaml}
-We present `baseCommand` as a two item list containing the command and the input string. CWL will combine these two items (in the order given) to make the full command when the script is run. This is not a complete tool descriptor yet, however - to find out what is missing we can use `cwltool` to validate the script:
-~~~
-$ cwltool --validate echo.cwl
-~~~
-{: .language-bash}
-~~~
-INFO .../cwltool 
 
-INFO Resolved 'echo.cwl' to 'file:///.../echo.cwl'
-ERROR Tool definition failed validation:
-echo.cwl:1:1: "outputs" section is not valid.
-~~~
-{: .output}
-`cwltool` has noted that we are missing an `outputs` section, so we will add this to our script:
-~~~
-cwlVersion: v1.2
-class: CommandLineTool
+The `cwlVersion` string defines which standard of the language is required for the tool or workflow. The most recent version is v1.2, and defaulting to this will enable your scripts to use all versions of the language, though some workflow engines which are not up-to-date may not run the script. This is, however, a hurdle to be tackled when you reach it.
```

Comment:
```suggestion
The `cwlVersion` string defines which standard of the language is required for the tool or workflow. The most recent version is v1.2.
```
All the major implementations support v1.2, so lets not worry about the earlier versions of CWL.